prereise/gather/demanddata/transportation_electrification/smart_charging.py

```diff
@@ -11,18 +11,38 @@
 )
 
 
+def ldv_weekend_weekend_check(x, y):
+    """Helper function to select weekday/weekend data rows
+
+    :param int x: data weekday/weekend value
+    :param int y: model year weekday/weekend value
+    :return: (*bool*) -- if data row matches whether the current day is a weekday/weekend
+    """
+    return x == y
+
+
+def hdv_use_all_data_rows(x, y):
+    """Helper function to select weekday/weekend data rows
+
+    :param int x: data weekday/weekend value
+    :param int y: model year weekday/weekend value
+    :return: (*bool*) -- always returns true to use all data rows
+    """
+    return True
```

Comment:
I don't quite follow what we are doing here.

Reply:
This looks similar to the `get_data_day` and/or `get_input_day' functions in `data_helper.py`.  Could one of those be used instead of creating a new function?

Reply:
Bainan - the context is that the LDV and LDT data from NHTS has a differentiation between weekday and weekend trip patterns.  The MDV and HDV data does not have that differentiation

Reply:
I see. Nonetheless, having a function simply returning `x == y` and another one always returning `True` seems to be redundant. I would suggest to use `lambda` if necessary.

Reply:
It's an intermediate refactoring step before changing the looping structure...I wanted to use lambda functions instead of function definitions but the linter complained. Ideally I'm going to get rid of the double loop over all the data every day, and switch out weekend and weekday dataframes for the light vehicle case.

The function `use_data_row` determines if a data row is used for a particular day:
``` python
if use_data_row(data_day[i], input_day[day_iter]):
```
and is set based on whether `veh_type` is "ldv"/"ldt" or "mdv"/"hdv"
``` python
  if veh_type.lower() == "ldv" or veh_type.lower() == "ldt":
      use_data_row = ldv_weekend_weekend_check
  elif veh_type.lower() == "mdv" or veh_type.lower() == "hdv":
      use_data_row = hdv_use_all_data_rows
```

Reply:
@dmuldrew So eventually we will get rid of this, right?

Reply:
Yep, I'm definitely not happy with it long-term

Reply:
Cool. Let's leave it as it is for now then.